.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -454,6 +464,16 @@ private void ExtendWithWeekdayAndYear(ref int startIndex, ref int endIndex, int
             // Check whether there's a weekday
             var prefix = text.Substring(0, startIndex);
             var matchWeekDay = this.Config.WeekDayEnd.Match(prefix);
+
+            // Check for weekday in the suffix
+            if (!matchWeekDay.Success)
+            {
+                suffix = text.Substring(endIndex);
+                var weekDayStart = @"^\s*,?\s*" + this.Config.WeekDayRegex;
+                Regex weekDayStartRegex = new Regex(weekDayStart, RegexOptions.Compiled | RegexOptions.IgnoreCase);

```

Comment:
@LionbridgeCSII, I made a mistake merging this. Sorry.
Regexes should never be constructed within methods that are called multiple times. Much less with flags like 'Compiled'. This change has caused a massive performance hit.
The correct approach is to introduce a new regex in the relevant config, or to modify regexes only during initialization. In this case, the solution would be to declare a WeekDayStart regex in YAML and config used here (similar scenario to WeekDayEnd referenced a couple lines earlier). 
For Turkish, this regex would be populated, for other languages, it would be kept empty or use a placeholder that never matches.
We'll push a fix soon. Please be aware of such changes in the future.

Reply:
ok, we will keep that in mind